modules/sor/sorV2/lib/poolsV2/weighted/weightedPoolToken.ts

```diff
@@ -0,0 +1,24 @@
+import { BigintIsh, Token, TokenAmount } from '@balancer/sdk';
+
+export class WeightedPoolToken extends TokenAmount {
+    public readonly weight: bigint;
+    public readonly index: number;
+
+    public constructor(token: Token, amount: BigintIsh, weight: BigintIsh, index: number) {
+        super(token, amount);
+        this.weight = BigInt(weight);
+        this.index = index;
+    }
+
+    public increase(amount: bigint): TokenAmount {
+        this.amount = this.amount + amount;
+        this.scale18 = this.amount * this.scalar;
+        return this;
+    }
+
+    public decrease(amount: bigint): TokenAmount {
+        this.amount = this.amount - amount;
+        this.scale18 = this.amount * this.scalar;
+        return this;
+    }
```

Comment:
I agree it would be a better naming, but since it's the pattern used on all pools, I don't think such a refactor fits in this PR. I created issue #682 to keep track of this.